src/sage/graphs/graph.py

```diff
@@ -6786,13 +6786,26 @@ def cliques_number_of(self, vertices=None, cliques=None):
             {(0, 0): 2, (0, 1): 3, (0, 2): 2, (1, 0): 2, (1, 1): 3, (1, 2): 2}
             sage: F.cliques_number_of(vertices=[(0, 1), (1, 2)])
             {(0, 1): 3, (1, 2): 2}
+            sage: F.cliques_number_of(vertices=(0, 1))
+            3
             sage: G = Graph({0:[1,2,3], 1:[2], 3:[0,1]})
             sage: G.show(figsize=[2,2])
             sage: G.cliques_number_of()
             {0: 2, 1: 2, 2: 1, 3: 1}
         """
-        import networkx
-        return networkx.number_of_cliques(self.networkx_graph(), vertices, cliques)
+        if cliques is None:
+            cliques = self.cliques_maximal()
+
+        if vertices in self:  # single vertex
+            return sum(1 for c in cliques if vertices in c)
+
+        from collections import Counter
+        count = Counter()
+
+        for c in cliques:
+            count.update(c)
+
+        return { v : count[v] for v in vertices or self }
```

Comment:
coding style: `return { v : count[v] for v in vertices or self }` -> `return {v : count[v] for v in vertices or self}`

Reply:
Done.